src/SleighAsm.cpp

```diff
@@ -17,30 +19,161 @@ void SleighAsm::init(RAsm *a)
 	if(!io)
 		throw LowlevelError("Can't get RIO from RBin");
 
+	initInner(io, a->cpu);
+}
+
+void SleighAsm::init(RAnal *a)
+{
+	if(description.empty())
```

Comment:
Too much code duplication here from `SleighAsm::init(RAsm *a)`. It will be better to merge in one function `SleighAsm::init(const char *sleigh_id, RIO *io, RConfig *cfg)`.